code/tools/ShortCodeRelationFinder.php

```diff
@@ -39,18 +39,22 @@ public function findPageCount($number)
     }
 
     /**
+     * @param int $number
      * @return DataList
      */
     public function getList($number)
     {
+        $number = (int) $number;
         $list = DataList::create('SiteTree');
         $where = array();
         $fields = $this->getShortCodeFields('SiteTree');
         $shortcode = DMS::inst()->getShortcodeHandlerKey();
         foreach ($fields as $ancClass => $ancFields) {
+            $ancClass = Convert::raw2sql($ancClass);
             foreach ($ancFields as $ancFieldName => $ancFieldSpec) {
+                $ancFieldName = Convert::raw2sql($ancFieldName);
                 if ($ancClass != "SiteTree") {
-                    $list = $list->leftJoin($ancClass, '"'.$ancClass.'"."ID" = "SiteTree"."ID"');
+                    $list = $list->leftJoin($ancClass, '"' . $ancClass . '"."ID" = "SiteTree"."ID"');
                 }
                 $where[] = "\"$ancClass\".\"$ancFieldName\" LIKE '%[{$shortcode},id=$number]%'"; //."%s" LIKE ""',
```

Comment:
Oh, you can't raw2sql on field identifiers; Only field values. SQL escaping isn't valid in the context you are trying to use them.

The "correct" behaviour is to error if identifiers contain double quotes.

Since those fields aren't injected, and come from the underlying model, they aren't unsafe to rely on.

Reply:
Ok, discussed and we'll leave this fix for now as it's pretty benign

Reply:
The only mandatory fix was `$number = (int) $number;` right? Do you still want to PR that?

Reply:
Yes

Reply:
PR at #204